### DISCUSSION TITLE
grid mode & external midi controllers

### DISCUSSION THREAD

**Discussion by duncanrmi** [SynthstromAudible/DelugeFirmware#2899] [General]
I'm sure it will turn out that there's a workaround, or that I was doing something wrong, but this weekend, on stage, I fell foul of this:
I have several clips set up for each of several instruments, all of them internal synths & kits.
I have been through the process of learning some external controllers (from an akai midimix think- standard midi CCs) to each of these instruments in the context of the clip (why can't the assignments be tied to the synth itself?)

then I discover- on stage!- that switching between clips of the same instrument DOESN'T hold the CC value. so volume jumps up & down, filters close & open..... 

I'm on a recent nightly- chopin- & also noticed that I'm missing the pink 'performance' button in grid mode too.

**Comment by seangoodvibes** [comment]
Use a midi takeover mode, eg value scaling, pickup, to avoid cc jumps

Perf view pink pad was moved to the keyboard button

**Comment by seangoodvibes** [comment]
If you clone a clip it should be keeping the midi cc learns

The midi learns are associated with the synth, not sure what you're saying on this one

**Comment by seangoodvibes** [comment]
Depending on the use case you might enjoy midi follow also (no learning required)

**Comment by RJ-Eckie** [comment]
The pink performance button has moved to the keyboard-button. I do really like it there, but I was also looking for it for a little bit when the pink pad was gone!

Did you happen to have any automation recorded on the parameters you had your CCs mapped to possibly?

**Comment by duncanrmi** [comment]
none of the things I was controlling from the midimix were recorded automation, no. 
I'll try these suggestions. but yes- I found that if I loaded the same synth into a different clip, I had to do the learning all over again.

I would far rather re-program the midimix (in as much as this is possible- some of the controls can't be edited) & use the deluge's 'hard-wired' CC assignments than the 'learn' function. I experienced what seemed like crosstalk between controllers, i.e. I'd alter a controller that I'd learned & something else unrelated would start changing.
